waspc/src/Wasp/Generator/AuthProviders/OAuth.hs

```diff
@@ -0,0 +1,64 @@
+module Wasp.Generator.AuthProviders.OAuth
+  ( frontendLoginUrl,
+    serverLoginUrl,
+    serverOauthRedirectHandlerUrl,
+    slug,
+    displayName,
+    logoFileName,
+    passportDependency,
+    scopeStr,
+    clientIdEnvVarName,
+    clientSecretEnvVarName,
+    OAuthAuthInfo (..),
+  )
+where
+
+import Data.Char (toUpper)
+import Data.List (intercalate)
+import StrongPath (File', Path', Rel')
+import Wasp.AppSpec.App.Dependency (Dependency)
+
+data OAuthAuthInfo = OAuthAuthInfo
+  { _slug :: String,
+    _displayName :: String,
+    _requiredScope :: OAuthScope,
+    _logoFileName :: Path' Rel' File',
```

Comment:
Changing this from `Path'` to `Path Posix` would cause some downstream changes since a lot of helper methods in the `Generator` are relying on system paths e.g.
 
 ```hs
asTmplFile :: Path' (Rel d) File' -> Path' (Rel WebAppTemplatesDir) File'
asTmplFile = SP.castRel
```

Reply:
I did a quick investigation: so we use this path both as System, to copy these files, and later as Posix, when we provide it to `<img src=` in `templates/react-app/src/auth/helpers/Generic.js`.

I think it is ok to keep it as System here, but we need to make sure it gets converted to Posix before giving it to `<img src=`.
That happens in file `WebAppGenerator/AuthG.hs`, in line 123 (`iconName=...`). `SP.toFilePath` is used there, but we should use `SP.fromRelFileP`. And you will want to use `SP.relFileToPosix` to convert it from System to Posix.